google-cloud-bigquery/acceptance/bigquery/dataset_test.rb

```diff
@@ -168,16 +170,26 @@
 
   it "imports data from a local file and creates a new table with specified schema in a block with load_job" do
     job_id = "test_job_#{SecureRandom.urlsafe_base64(21)}" # client-generated
-    job = dataset.load_job "local_file_table", local_file, job_id: job_id do |schema|
-      schema.integer   "id",    description: "id description",    mode: :required
-      schema.string    "breed", description: "breed description", mode: :required
-      schema.string    "name",  description: "name description",  mode: :required
-      schema.timestamp "dob",   description: "dob description",   mode: :required
+    job = dataset.load_job "local_file_table", local_file, job_id: job_id do |job|
+      job.schema.integer   "id",    description: "id description",    mode: :required
+      job.schema.string    "breed", description: "breed description", mode: :required
+      job.schema.string    "name",  description: "name description",  mode: :required
+      job.schema.timestamp "dob",   description: "dob description",   mode: :required
+      job.schema_update_options = schema_update_options
+      job.time_partitioning_type = "DAY"
```

Comment:
What do you think about creating a method that takes these 4 values as arguments and represents them as an object instead of exposing 5 prefixed values?

Reply:
I'd be happy to do that if you think it's better. Our past design has been to flatten, but I agree that the number of methods makes it seem reasonable to have a value object.

Reply:
I don't think we've been 100% consistent. In some other gems like Storage we would have a method that sets all the values, and an object that represents the collected values. I think it makes sense here, as all these values are dependent on each other.